packages/datetime/src/dateRangeInput.tsx

```diff
@@ -13,21 +14,43 @@ import {
     IInputGroupProps,
     InputGroup,
     IProps,
+    Popover,
+    Position,
     Utils,
 } from "@blueprintjs/core";
 
 import {
+    DateRange,
+    fromDateToMoment,
+    fromMomentToDate,
+    isMomentValidAndInRange,
+} from "./common/dateUtils";
+import {
+    getDefaultMaxDate,
+    getDefaultMinDate,
     IDatePickerBaseProps,
 } from "./datePickerCore";
+import {
+    DateRangePicker,
+} from "./dateRangePicker";
 
 export interface IDateRangeInputProps extends IDatePickerBaseProps, IProps {
-    startInputProps?: IInputGroupProps;
     endInputProps?: IInputGroupProps;
+    onChange?: (selectedRange: DateRange) => void;
+    startInputProps?: IInputGroupProps;
 }
 
-export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, {}> {
+export interface IDateRangeInputState {
+    isOpen?: boolean;
+    selectedEnd?: moment.Moment;
+    selectedStart?: moment.Moment;
```

Comment:
optional states make the usage much easier (so you can supply partial objects, which React merges with `this.state`).

with `strictNullChecks` we'd be able to make all state fields required, but there are other gotchas with that change so we haven't been able to do that yet.